packaging/container/Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM rockylinux/rockylinux:8
+LABEL maintainer="tdockendorf@osc.edu; johrstrom@osc.edu"
+
+ARG VERSION=3.1
+
+# setup the ondemand repositories
+RUN dnf -y install https://yum.osc.edu/ondemand/${VERSION}/ondemand-release-web-${VERSION}-1.el8.noarch.rpm
+
+# install all the dependencies
+RUN dnf -y update && \
+    dnf install -y dnf-utils epel-release && \
+    dnf config-manager --set-enabled powertools && \
+    dnf -y module enable nodejs:18 ruby:3.1 && \
+    dnf install -y \
+        ondemand \
+        ondemand-dex && \
+    dnf clean all && rm -rf /var/cache/dnf/*
+
+COPY launch-ood      /opt/ood/launch
+
+# copy configuration files
+RUN mkdir -p /etc/ood/config
+RUN cp /opt/ood/nginx_stage/share/nginx_stage_example.yml            /etc/ood/config/nginx_stage.yml
+RUN cp /opt/ood/ood-portal-generator/share/ood_portal_example.yml    /etc/ood/config/ood_portal.yml
+
+# make some misc directories & files
+RUN mkdir -p /var/lib/ondemand-nginx/config/apps/{sys,dev,usr}
+RUN touch /var/lib/ondemand-nginx/config/apps/sys/{dashboard,shell,myjobs}.conf
+
+# setup sudoers for apache
+RUN echo -e 'Defaults:apache !requiretty, !authenticate \n\
+Defaults:apache env_keep += "NGINX_STAGE_* OOD_*" \n\
+apache ALL=(ALL) NOPASSWD: /opt/ood/nginx_stage/sbin/nginx_stage' >/etc/sudoers.d/ood
+
+# run the OOD executables to setup the env
+RUN /opt/ood/ood-portal-generator/sbin/update_ood_portal --insecure
```

Comment:
Should we still install from source and use `--insecure` for this? This is going to have a lot of caveats to be actually useful, but I feel like a package install and a systemd entypoint is really the _only_ way anyone will be able to use this in a meaningful way.

Reply:
Installing from source is not a supported way to install OnDemand so want to avoid that.  This container will be far from useful irregardless of the install mechanisms.  The insecure is because there are no SSL certs I believe.

systemd is not really used in traditional containers.  That's essentially treating a container like a VM if you launch it with init.

Reply:
Sorry for the delay - I started a review or something and my comments were never posted.

> Installing from source is not a supported way to install OnDemand so want to avoid that.

This Dockerfile installs from source.

>  This container will be far from useful irregardless of the install mechanisms.

This is kind of why I've hesitated to implement this at all. Why not something like this that just installs OnDemand and call it a day? No entrypoint, it just has everything installed and they have to take it further.

```Dockerfile
FROM rockylinux/rockylinux:8

RUN dnf install -y https://yum.osc.edu/ondemand/3.1/ondemand-release-web-3.1-1.el8.noarch.rpm && \
    dnf clean all && rm -rf /var/cache/dnf/*

RUN dnf -y update && \
    dnf install -y dnf-utils && \
    dnf config-manager --set-enabled powertools && \
    dnf -y module enable nodejs:18 ruby:3.1 && \
    dnf install -y epel-release && \
    dnf install -y ondemand ondemand-dex && \
    dnf install -y gcc gcc-c++ libyaml-devel nc && \
    dnf clean all && rm -rf /var/cache/dnf/*
```

> systemd is not really used in traditional containers. That's essentially treating a container like a VM if you launch it with init.

Yet another reason I don't really want to do this. If you launch apache from a script can you bounce it? Or maybe they'd have to bounce the whole container.

Reply:
You bounce the whole container, but our entrypoint is an example, not what folks should use.  I only use it to validate the container works but we also use the same entry point for dev containers.  It's a nice way to validate OnDemand container works but not something folks should be using.  It's trivial to override ENTRYPOINT with site specific Dockerfile that pulls from our image.